src/Chapter3.hs

```diff
@@ -971,6 +1084,29 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data DayOfWeek
+  = Monday
+  | Tuesday
+  | Wednesday
+  | Thursday
+  | Friday
+  | Saturday
+  | Sunday
+  deriving (Show, Enum)
+
+isWeekend :: DayOfWeek -> Bool
+isWeekend d = fromEnum d >= 5
+
+nextDay :: DayOfWeek -> DayOfWeek
+nextDay Sunday = Monday
+nextDay d = succ d
+
+daysToParty :: DayOfWeek -> Int
+daysToParty d = if result >= 0
+                then result
+                else 7 + result
+                where result = 4 - fromEnum d
```

Comment:
You can use `mod` function in here instead 🙂 

Also, a small styling suggestion. Usually it is not that convenient to make the indentation depend on the previous lines. In here if you change the name of the function or a variable name you would need to realign everything..
`where` keyword is usually written on a separate line with 2 spaces, and all expressions under where are starting on the next line with 4 spaces:

```haskell
daysToParty d = 
    if result >= 0
    then result
    else 7 + result
  where 
    result = 4 - fromEnum d
```